Terminal.Gui/ConsoleDrivers/NetDriver.cs

```diff
@@ -1363,37 +1351,61 @@ public override void ResizeScreen ()
 							Console.CursorLeft = 0;
 							Console.WindowTop = 0;
 							Console.WindowLeft = 0;
+							if (Console.WindowHeight > Rows) {
+								Console.SetWindowSize (Cols, Rows);
+							}
 							Console.SetBufferSize (Cols, Rows);
 #pragma warning restore CA1416
 						} else {
 							//Console.Out.Write ($"\x1b[8;{Console.WindowHeight};{Console.WindowWidth}t");
 							Console.Out.Write ($"\x1b[0;0" +
-								$";{Rows};{Cols}w");
+							    $";{Rows};{Cols}w");
 						}
 					} catch (System.IO.IOException) {
-						return;
+						setClip ();
 					} catch (ArgumentOutOfRangeException) {
-						return;
+						setClip ();
 					}
 				}
 			} else {
 				if (IsWinPlatform && Console.WindowHeight > 0) {
 					// Can raise an exception while is still resizing.
 					try {
 #pragma warning disable CA1416
-						Console.WindowTop = Math.Max (Math.Min (top, Rows - Console.WindowHeight), 0);
+						Console.CursorTop = 0;
+						Console.CursorLeft = 0;
+						//Console.WindowTop = 0;
+						//Console.WindowLeft = 0;
+						if (Console.WindowHeight > Rows) {
+							Console.SetWindowSize (Cols, Rows);
+						}
+						Console.SetBufferSize (Cols, Rows);
 #pragma warning restore CA1416
-					} catch (Exception) {
-						return;
+					} catch (System.IO.IOException) {
+						setClip ();
+					} catch (ArgumentOutOfRangeException) {
+						setClip ();
 					}
 				} else {
 					Console.Out.Write ($"\x1b[{top};{Console.WindowLeft}" +
-						$";{Rows};{Cols}w");
+					    $";{Rows};{Cols}w");
+				}
+			}
+			setClip ();
+
+			void setClip ()
```

Comment:
Can you explain what this does and why you think "[3J" is the correct behavior? Intuitively, it seems "the less destructive choice is always better" and thus "[0J" or "[1J" would be better (because I think "3J" clears everything in the console that was there before the app ran).

Reply:
While the new values are processing the window may been resizing again at the same time, throwing exceptions, but is more accurate do set the Clip with the available values than return from the without setting. I didn't see the need of doing an external method for that and decided for a local method.
Yes, it's more destructive and that why I used it to ensure that no buffer exist if `HeightAsBuffer` is false. This code is only executed after the app is running and only affect the app's contents. To preserve the original console contents there are code to save in the `Init` method and to restore on the `End` method. I realizing that this isn't needed any more, so I'll comment it for now.